File: mobile/retrieveAirQualityData.dart
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';

class AirQualityList extends StatefulWidget {
  @override
  _AirQualityListState createState() => _AirQualityListState();
}

class _AirQualityListState extends State<AirQualityList> {
  List<Map<String, dynamic>> data = [];
  final String token = 'your_access_token'; // Replace with the access token

  @override
  void initState() {
    super.initState();
    fetchData();
  }

  Future<void> fetchData() async {
    final String siteId = 'your_site_id'; // Replace with the Site ID for City X
    final String apiUrl =
        'https://api.airqo.net/api/v2/devices/measurements/sites/$siteId?token=$token';

    final response = await http.get(Uri.parse(apiUrl));
    if (response.statusCode == 200) {
      final List<dynamic> jsonData = json.decode(response.body)['data'];
      setState(() {
        data = jsonData.cast<Map<String, dynamic>>();
      });
    } else {
      throw Exception('Failed to load data');
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('City X Air Quality Data'),
      ),
      body: ListView.builder(
        itemCount: data.length,
        itemBuilder: (context, index) {
          final measurement = data[index];
          return ListTile(
            title: Text('Date: ${measurement['date']}'),
            subtitle: Text('PM2.5: ${measurement['pm25']}'),
            // Include more data fields as needed
          );
        },
      ),
    );
  }
}

File: javascript/retrieveAirQualityData.js
import React, { useState, useEffect } from "react";

const AirQualityData = () => {
  const [data, setData] = useState([]);
  const [token, setToken] = useState(""); // Input field for the access token

  const gridId = "your_grid_id"; // Replace with the grid ID for City X

  useEffect(() => {
    // Make the API request
    const fetchData = async () => {
      try {
        const response = await fetch(
          `https://api.airqo.net/api/v2/devices/measurements/grids/${gridId}?token=${token}`
        );
        if (response.ok) {
          const result = await response.json();
          setData(result.data);
        } else {
          console.error("Error fetching data");
        }
      } catch (error) {
        console.error("Error:", error);
      }
    };

    if (token && gridId) {
      fetchData();
    }
  }, [token, gridId]);

  return (
    <div>
      <h1>City X Air Quality Data</h1>
      <p>Input your access token:</p>
      <input
        type="text"
        value={token}
        onChange={(e) => setToken(e.target.value)}
      />
      <div>
        {data.map((measurement) => (
          <div key={measurement.id}>
            <p>Date: {measurement.date}</p>
            <p>PM2.5: {measurement.pm25}</p>
            {/* Include more data fields as needed */}
          </div>
        ))}
      </div>
    </div>
  );
};

export default AirQualityData;
